src/bilder/lib/template_helpers.py

```diff
@@ -0,0 +1,133 @@
+from pathlib import Path
+
+from pyinfra.operations import files
+
+from bilder.components.hashicorp.consul_template.models import ConsulTemplateTemplate
+
+CONSUL_TEMPLATE_DIRECTORY = "/etc/consul-template"
+
+
+def place_jinja_template_file(  # noqa: WPS211
+    name: str,
+    repo_path: Path,
+    destination_path: Path,
+    context: dict,
+    watched_files: list[Path],
```

Comment:
I think that it's best if we don't mutate the `watched_files` as part of the body of the function. What might work better is if the return value of the function is the destination path of the file on disk, and that can then be used in a pattern like:
```
watched_files.append(place_jinja_template_file(...))
```